dedalus/core/basis.py

```diff
@@ -4969,4 +5051,83 @@ def operate(self, out):
         np.copyto(out.data, operand.data[axslice(self.index,0,2)])
 
 
+class CartesianAdvectiveCFL(operators.AdvectiveCFL):
+
+    input_coord_type = CartesianCoordinates
+    input_basis_type = (ComplexFourier, RealFourier, Jacobi)
+
+    @CachedMethod
+    def grid_spacing(self, velocity):
+        coordsys = velocity.tensorsig[0]
+        grid_spacing = np.zeros_like(velocity.data)
+        for i, c in enumerate(coordsys.coords):
+            basis = velocity.domain.get_basis(c)
+            global_spacing = basis.grid_spacing(scale=basis.dealias[0])
```

Comment:
If the bases all provide `global_grid_spacing`, maybe they should also provide `local_grid_spacing` instead of doing the slicing here?